spanner/google/cloud/spanner_v1/types.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2017, Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+import sys
+
+from google.gax.utils.messages import get_messages
+
+from google.api import auth_pb2
+from google.api import http_pb2
+from google.cloud.proto.spanner.v1 import keys_pb2
+from google.cloud.proto.spanner.v1 import mutation_pb2
+from google.cloud.proto.spanner.v1 import query_plan_pb2
+from google.cloud.proto.spanner.v1 import result_set_pb2
+from google.cloud.proto.spanner.v1 import spanner_pb2
+from google.cloud.proto.spanner.v1 import transaction_pb2
+from google.cloud.proto.spanner.v1 import type_pb2
+from google.protobuf import descriptor_pb2
+from google.protobuf import duration_pb2
+from google.protobuf import empty_pb2
+from google.protobuf import struct_pb2
+from google.protobuf import timestamp_pb2
+
+names = []
+for module in (auth_pb2, http_pb2, keys_pb2, mutation_pb2, query_plan_pb2,
+               result_set_pb2, spanner_pb2, transaction_pb2, type_pb2,
+               descriptor_pb2, duration_pb2, empty_pb2, struct_pb2,
+               timestamp_pb2, ):
+    for name, message in get_messages(module).items():
+        message.__module__ = 'google.cloud.spanner_v1.types'
+        setattr(sys.modules[__name__], name, message)
+        names.append(name)
```

Comment:
I can't say I like this monkeying with the original module of the protos.  For one thing, we aren't documenting them here, and so hiding the real source makes it harder for users to find the original docs.

Reply:
> I can't say I like this monkeying with the original module of the protos. For one thing, we aren't documenting them here, and so hiding the real source makes it harder for users to find the original docs.

I am definitely trying to move toward documenting `types` consistently as `types`. It is much more usable than `random_thing_pb2`.